[16.0][ADD] account_invoice_mass_sending_direct_print

thienvh332 opened the pull request:
This module only affects the buttons below, other tasks will not be affected.
![image](https://github.com/user-attachments/assets/68269cd1-7dc0-4c7b-b075-1af1f8729e77)

![image](https://github.com/user-attachments/assets/2b52bde9-9d65-4eba-8ba6-5b5f5831d311)

![image](https://github.com/user-attachments/assets/542bf37e-214a-4319-a401-f619be6fce31)

thienvh332 committed:
[ADD] account_invoice_mass_sending_direct_print

nilshamerlinck commented on account_invoice_mass_sending_direct_print/wizards/account_invoice_send.py:
```diff
@@ -0,0 +1,77 @@
+class AccountInvoiceSend(models.TransientModel):
+    _inherit = "account.invoice.send"
+
+    is_direct_print = fields.Boolean(default=False, string="Direct Print")
```
can you please introduce a `default=lambda self: self.env.company.invoice_is_direct_print`, also shown with settings as `invoice_print`

nilshamerlinck commented on account_invoice_mass_sending_direct_print/models/res_config_settings.py:
```diff
@@ -0,0 +1,11 @@
+        string="Invoice Is Direct Print",
+        related='company_id.invoice_is_direct_print',
+        readonly=False,
+        help="Allow invoice to be printed directly when 'Send and Print' is selected",
```
selected or clicked?

nilshamerlinck commented on account_invoice_mass_sending_direct_print/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1,7 @@
+Account Invoice Mass Sending Direct Print
+==========================================
+
+This module extends the functionality of the `account_invoice_mass_sending` module by enabling direct printing of invoices. 
```
```suggestion
This module extends the functionality of the `account_invoice_mass_sending` module to enable direct printing of invoices. 
```

nilshamerlinck commented:
- screenshots?
- is only the button `Send and Print` impacted or also the other one?

thienvh332 commented:
> * screenshots?
> * is only the button `Send and Print` impacted or also the other one?

I updated description. Thanks sir

nilshamerlinck commented on account_invoice_mass_sending_direct_print/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1,7 @@
+==========================================
+
+This module extends the functionality of the `account_invoice_mass_sending` module to enable direct printing of invoices. 
+
+Key Features:
+- Automatically send invoices to a configured printer during mass sending operations.
```
confusing
you can remove
what if user only prints and doesn't send?

nilshamerlinck commented on account_invoice_mass_sending_direct_print/models/res_config_settings.py:
```diff
@@ -0,0 +1,12 @@
+        string="Invoice Is Direct Print",
+        related="company_id.invoice_is_direct_print",
+        readonly=False,
+        help="Allow invoice to be printed directly when 'Send and Print' is clicked",
```
confusing

better say things explicitly then if you are not referring to a specific button: the `Send & Print` wizard

sergiobstoj committed:
[FIX] account_invoice_mass_sending: Send as comment to preserve mail tracking and keep avoiding extra notifications

jonpasquier committed:
Added translation using Weblate (Catalan)

jonpasquier committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (25 of 25 strings)

Translation: account-invoicing-16.0/account-invoicing-16.0-account_invoice_google_document_ai
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-16-0/account-invoicing-16-0-account_invoice_google_document_ai/ca/

jonpasquier committed:
Added translation using Weblate (Catalán (ca_ES) (ca_ES))

nilshamerlinck committed:
[FIX] pin cryptography<37 for pyopenssl 20.0.1 compatibility

google-cloud-documentai (dep of account_invoice_google_document_ai) pulls in
google-auth, which upgrades cryptography beyond what pyopenssl 20.0.1 supports.
pyopenssl 20.0.1 uses internal cryptography bindings (X509_V_FLAG_CB_ISSUER_CHECK,
X509_V_FLAG_NOTIFY_POLICY) that were removed in cryptography 37.0.0.

nilshamerlinck committed:
[FIX] pin cryptography<37 for pyopenssl 20.0.1 compatibility

google-cloud-documentai (dep of account_invoice_google_document_ai) pulls in
google-auth, which upgrades cryptography beyond what pyopenssl 20.0.1 supports.
pyopenssl 20.0.1 uses internal cryptography bindings (X509_V_FLAG_CB_ISSUER_CHECK,
X509_V_FLAG_NOTIFY_POLICY) that were removed in cryptography 37.0.0.

pedrobaeza committed:
Merge pull request #2257 from nilshamerlinck/16.0-fix-deps

[16.0][FIX] pin cryptography<37 for pyopenssl 20.0.1 compatibility

OCA-git-bot committed:
Merge PR #2255 into 16.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update account_invoice_mass_sending.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-invoicing-16.0/account-invoicing-16.0-account_invoice_mass_sending
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-16-0/account-invoicing-16-0-account_invoice_mass_sending/
